src/hotspot/share/gc/g1/g1MonitoringSupport.hpp

```diff
@@ -239,11 +239,32 @@ class G1MonitoringSupport : public CHeapObj<mtGC> {
 // Scope object for java.lang.management support.
 class G1MonitoringScope : public StackObj {
   G1MonitoringSupport* _monitoring_support;
+public:
+  G1MonitoringScope(G1MonitoringSupport* monitoring_support);
+  ~G1MonitoringScope();
+};
+
+class G1IncMonitoringScope : public G1MonitoringScope {
   TraceCollectorStats _tcs;
   TraceMemoryManagerStats _tms;
 public:
-  G1MonitoringScope(G1MonitoringSupport* monitoring_support, bool full_gc, bool all_memory_pools_affected);
-  ~G1MonitoringScope();
+  G1IncMonitoringScope(G1MonitoringSupport* monitoring_support, bool all_memory_pools_affected);
+  ~G1IncMonitoringScope() { }
+};
+
+class G1FullMonitoringScope : public G1MonitoringScope {
```

Comment:
```suggestion
class G1FullGCMonitoringScope : public G1MonitoringScope {
```
Similiarly, the prefix should be "FullGC" for consistency.